core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -155,9 +155,12 @@ public RewriteStrategy options(Map<String, String> options) {
   public Iterable<List<FileScanTask>> planFileGroups(Iterable<FileScanTask> dataFiles) {
     ListPacker<FileScanTask> packer = new BinPacking.ListPacker<>(maxGroupSize, 1, false);
     List<List<FileScanTask>> potentialGroups = packer.pack(dataFiles, FileScanTask::length);
+
     return potentialGroups.stream().filter(group ->
-      group.size() >= minInputFiles || sizeOfInputFiles(group) > targetFileSize ||
-              group.stream().anyMatch(this::taskHasTooManyDeletes)
+            (group.size() >= minInputFiles && group.size() > 1) ||
+                sizeOfInputFiles(group) > targetFileSize ||
+                group.stream().anyMatch(this::taskHasTooManyDeletes) ||
+                (group.stream().anyMatch(this::taskHasDeletes) && group.size() == 1)
```

Comment:
Why do we need this check? Shouldn't "taskHasToManyDeletes" be the catch here?

Reply:
I am thinking of a case like Group with (1 Data file with 1 delete) with deleteFileThreshold = 2
3 as per my understanding would return false I beleive as deletes  = 1 which is < 2
```java
  private boolean taskHasTooManyDeletes(FileScanTask task) {
    return task.deletes() != null && task.deletes().size() >= deleteFileThreshold;
  }
```

4 as per my understanding would return return true  as delete = 1 which >= 1
```java
  private boolean taskHasDeletes(FileScanTask task) {
    return task.deletes() != null && task.deletes().size() >= 1;
  }
```

Appologies seems like I am missing something here, can you please elaborate more